src/rest_api/classes/laboratory/widgets/members.clj

```diff
@@ -1,31 +1,108 @@
 (ns rest-api.classes.laboratory.widgets.members
   (:require
-    [rest-api.classes.generic-fields :as generic]
+    [datomic.api :as d]
+    [clojure.string :as str]
+    [pseudoace.utils :as pace-utils]
+    [rest-api.formatters.date :as date]
     [rest-api.formatters.object :as obj :refer [pack-obj]]))
 
-(defn- get-lineage-data [m]
- ;; supervised-by supervised worked-with
- (let []
-  {:name (pack-obj m)
-   :k (keys m)
-   :d (:db/id m)
-   :level nil
-   :start_date nil
-   :end_date nil
-   :duration nil
-   }
-  ))
+(defn- generate-map [oid]
+  (let [other-person-id (get oid 0)
+        other-person-name (get oid 1)
+        json-data (get oid 2)
+        roles (filter #(re-find #"role" (str %)) (keys json-data))
+        level
+        (for [role roles
+              :let
+              [level (-> role
+                         (str/replace #":role/" "")
+                         (str/replace #"-" " ")
+                         (str/capitalize))]] level)
```

Comment:
This part could be:
(some->> roles
                (map name)
                (map obj/humanize-ident))

Reply:
Thanks, I see that works, and it's good to know about obj/humanize-ident .
I see that it works without the (map name), (name) is the first thing applied by obj/humanize-ident .